### ISSUE TITLE
Hash character in Markdown code span not rendered correctly

### ISSUE THREAD

**Issue by c-lipka** [doxygen/doxygen#6840]
Hash characters (`#`) in Markdown code spans are not rendered correctly when immediately followed by a word: If the hash character is immediately preceded by a word, it is rendered like a C++ namespace separator (`::`), otherwise it is simply swallowed.

Examples:

~~~
`#if` is rendered like `if`.
`foo#bar` is rendered like `foo::bar`.
`foo #bar` is rendered like `foo bar`.
`foo # bar` is rendered ok.
`foo#` is rendered ok.
~~~

Observed with Doxygen 1.8.15, with documentation placed in a Markdown special page file.

**Comment by albert-github** [comment]
Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by c-lipka** [comment]
The issue seems to be sensitive to the `AUTOLINK_SUPPORT` setting: With that setting to `YES` (default), the cases where the hash is swallowed are fine; however, the cases where it is replaced with `::` (`foo#bar` in the above examples) is still broken.

I'll submit a minimal example shortly.

**Comment by c-lipka** [comment]
Demonstrates both this and #6054:

[doxygen-issues-6054-6840.zip](https://github.com/doxygen/doxygen/files/2869771/doxygen-issues-6054-6840.zip)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6842

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
